pubsub/message.go

```diff
@@ -59,19 +53,23 @@ type Message struct {
 	// size is the approximate size of the message's data and attributes.
 	size int
 
-	calledDone bool
-
-	// The done method of the iterator that created this Message.
-	doneFunc func(string, bool, time.Time)
-
 	// OrderingKey identifies related messages for which publish order should
 	// be respected. If empty string is used, message will be sent unordered.
 	OrderingKey string
+
+	// ackh handles Ack() or Nack().
+	ackh ackHandler
+}
+
+// NewMessage creates a message with a custom ack/nack handler, which should not
+// be nil.
+func NewMessage(ackh ackHandler) *Message {
```

Comment:
This is a poor API. It'll look weird in package docs since `ackHandler` is unexported.

Reply:
I don't see why `ackHandler` can't be exported. Maybe that should be the quick fix?

We also technically haven't cut a release for this commit, so I think we can still make some changes as a last-ditch effort if this doesn't sit well with you.

Reply:
The package docs looking strange is a good point. I've opened https://github.com/googleapis/google-cloud-go/pull/3216.